pkg/vmimage/ssh.go

```diff
@@ -11,11 +11,76 @@ import (
 )
 
 func (builder *Builder) ssh() error {
-	signer, err := ssh.NewSignerFromKey(builder.SSHKey)
+	session, err := builder.newSSHSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	stdout, err := session.StdoutPipe()
 	if err != nil {
 		return err
 	}
 
+	err = session.Start("sudo tail -F -n +1 /var/lib/waagent/custom-script/download/0/stdout")
+	if err != nil {
+		return err
+	}
+
+	io.Copy(os.Stdout, stdout)
+
+	return nil
+}
+
+func (builder *Builder) scp(files []string) error {
```

Comment:
can we share some of the logic between `*Builder.ssh()` and `*Builder.scp(files []string)`?

Reply:
We already doing session creation. The only other thing left is `stdoutPipe` but not sure if it worth start passing it around.